### ISSUE TITLE
"Expected 1 scheduled action under StartInstances, found 2" while creating 2x aws_appautoscaling_scheduled_action for aws_spot_fleet_request

### ISSUE THREAD

**Issue by voodoodror** [terraform-providers/terraform-provider-aws#12615]
### Terraform Version
core 0.12.18
provider version 2.41.0

### Affected Resource(s)

* aws_appautoscaling_scheduled_action.StartInstances
* aws_appautoscaling_scheduled_action.StopInstances

### Terraform Configuration Files

```hcl
resource "aws_spot_fleet_request" "spot-fleet" {
  iam_fleet_role      = "arn:aws:iam::XXXXXXXXX:role/aws-ec2-spot-fleet-tagging-role"
  allocation_strategy = local.allocation_strategy
  target_capacity     = 0

  valid_until = "2021-04-01T16:02:33Z"

  load_balancers = local.load_balancers
  target_group_arns = local.target_group_arns
  
  launch_specification {
    instance_type             = "m5.xlarge"
    ami                       = local.ami
    key_name                  = local.key_name
    subnet_id                 = "subnet-XXXXXXXXXXX"
    vpc_security_group_ids    = local.vpc_security_group_ids
    
    ebs_optimized             = local.ebs_optimized
    iam_instance_profile_arn  = data.aws_iam_instance_profile.instance-profile.arn
    root_block_device {
        volume_size           = local.root_block_device.volume_size
        volume_type           = local.root_block_device.volume_type
    }
    monitoring                = local.scaling["enabled"] == true ? true : local.monitoring
    tags = local.tags
  }
}

resource "aws_appautoscaling_target" "spot_fleet" {
  count              = local.scheduled["enabled"] == true || local.scaling["enabled"] == true ? 1 : 0
  min_capacity       = local.scaling["min_capacity"]
  max_capacity       = local.scaling["max_capacity"]
  resource_id        = "spot-fleet-request/${aws_spot_fleet_request.spot-fleet.id}"
  scalable_dimension = "ec2:spot-fleet-request:TargetCapacity"
  service_namespace  = "ec2"

  depends_on         = [aws_spot_fleet_request.spot-fleet]
}


resource "aws_appautoscaling_scheduled_action" "StartInstances" {
  count              = local.scheduled["enabled"] == true ? 1 : 0
  name               = "StartInstances"
  service_namespace  = "ec2"
  resource_id        = "spot-fleet-request/${aws_spot_fleet_request.spot-fleet.id}"
  scalable_dimension = "ec2:spot-fleet-request:TargetCapacity"
  schedule           = local.scheduled["start_cron"]

  scalable_target_action {
    min_capacity = local.scheduled["start_capacity"]
    max_capacity = local.scheduled["start_capacity"]
  }
  depends_on = [aws_spot_fleet_request.spot-fleet, aws_appautoscaling_target.spot_fleet[0]]
}

resource "aws_appautoscaling_scheduled_action" "StopInstances" {
  count              = local.scheduled["enabled"] == true ? 1 : 0
  name               = "StopInstances"
  service_namespace  = "ec2"
  resource_id        = "spot-fleet-request/${aws_spot_fleet_request.spot-fleet.id}"
  scalable_dimension = "ec2:spot-fleet-request:TargetCapacity"
  schedule           = local.scheduled["stop_cron"]

  scalable_target_action {
    min_capacity = 0
    max_capacity = 0
  }
  depends_on = [aws_spot_fleet_request.spot-fleet, aws_appautoscaling_target.spot_fleet[0]]
}

```

### Panic Output

```

Error: Expected 1 scheduled action under StartInstances, found 2

  on us-east-1.tf line 289, in resource "aws_appautoscaling_scheduled_action" "StartInstances":
 289: resource "aws_appautoscaling_scheduled_action" "StartInstances" {

```
### Expected Behavior

Create both scheduled actions without errors.

### Actual Behavior

It creates both scheduled actions but there's an error.

### Steps to Reproduce

`terraform apply`

**Comment by bflad** [comment]
Fix submitted here: https://github.com/terraform-providers/terraform-provider-aws/pull/12699

In the meantime, using unique names across all scheduled actions of the same service namespace should workaround the issue.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.60.0 of the Terraform AWS Provider, later this week. 👍
